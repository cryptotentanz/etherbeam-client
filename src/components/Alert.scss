@use '../styles/resources/sizes';

.my-alert {
  border-width: sizes.$border;
  border-style: solid;
  border-radius: sizes.$border-radius;
  padding: sizes.$spacer-s;
}
