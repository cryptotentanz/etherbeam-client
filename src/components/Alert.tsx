import React, { FC, ReactNode } from 'react'
import './Alert.scss'

type ColorProp = 'dark' | 'light' | 'primary' | 'secondary' | 'positive' | 'negative'

interface AlertProps {
  color?: ColorProp
  children?: ReactNode
}

const Alert: FC<AlertProps> = ({ color = 'light', children }: AlertProps) => {
  return <div className={`my-alert my-${color}-bg`}>{children}</div>
}

export default Alert
