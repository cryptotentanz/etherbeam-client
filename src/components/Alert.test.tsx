import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Alert from './Alert'

describe('<Alert />', () => {
  const content = 'Alert.'
  let component: ShallowWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = shallow(<Alert>{content}</Alert>)
    })

    it('should render', () => {
      const div = component.find('div').at(0)
      expect(div.hasClass('my-alert')).toBeTruthy()
      expect(div.hasClass('my-light-bg')).toBeTruthy()
      expect(div.prop('children')).toBe(content)
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('Color', () => {
    beforeAll(() => {
      component = shallow(<Alert color="primary">{content}</Alert>)
    })

    it('should render', () => expect(component.find('div').at(0).hasClass('my-primary-bg')).toBeTruthy())
  })
})
