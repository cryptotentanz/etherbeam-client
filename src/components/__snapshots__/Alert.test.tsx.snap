// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> Default should match snapshot 1`] = `
<div
  class="my-alert my-light-bg"
>
  Alert.
</div>
`;
